azurerm/resource_arm_application_gateway.go

```diff
@@ -499,7 +504,7 @@ func resourceArmApplicationGateway() *schema.Resource {
 
 						"host": {
 							Type:     schema.TypeString,
-							Required: true,
+							Optional: true,
```

Comment:
Now that this is optional we should probably add validation somewhere to make sure if this is empty, `pick_host_name_from_backend_address` is true.

Reply:
We should also update the documentation with this change

Reply:
@mdsills, in the create function you can check both properties and if neither are set return an error.

Something like:
```go
host := v["pick_host_name_from_backend_address"].(string)
pick := v["pick_host_name_from_backend_address"].(bool)
if !pick && host == "" {
return fmt.Errorf("one of `host` or `pick_host_name_from_backend_http_settings` must be set")
}

if pick && host != "" {
return fmt.Errorf("Only one of `host` or `pick_host_name_from_backend_http_settings` must be set")
}
```

disclaimer that code might not compile 😅

Reply:
thanks for the pointer. it's ended up looking rather different, but hopefully still acceptable, or at least a step towards acceptable...